### PR TITLE
[stable/datadog]add RBAC support

### PR DESCRIPTION
**Pull request by cw-sakamoto**
I checked the necessary role by looking at the following:
https://docs.datadoghq.com/integrations/kubernetes/

### CODE REVIEW

**Comment by cw-sakamoto** [comment]
@mgoodness @viglesiasce @dhilipkumars Hi, thanks for maintaining charts 👍
I can not use rbac in datadog chart and I am in trouble now, would you mind reviewing this early? 😢

**Comment by cw-sakamoto** [comment]
@dhilipkumars I appreciate your quick response. Thank you.

File: stable/datadog/values.yaml [hunk 1]
```diff
@@ -146,3 +146,9 @@ datadog:
     limits:
       cpu: 256m
       memory: 512Mi
+
+rbac:
+  create: false
```
Comment (dhilipkumars) [inline stable/datadog/values.yaml]: why is it false by default? could you add more comments on the new entries in `values.yaml`?

Here is a reference [PR](https://github.com/coreos/alb-ingress-controller/pull/146/files) and reference [issue](https://github.com/kubernetes/helm/issues/3098)

Comment (cw-sakamoto) [reply stable/datadog/values.yaml]: @dhilipkumars Thanks for your review 😄 
Sorry, I didn't check [issue](https://github.com/kubernetes/helm/issues/3098).
I set this value to ```false```, most charts seem to be defaulting to false.

But, I agree on that issue and I changed default value to ```true``` for ```rbac.create```  in ```values.yaml``` 👍

**Comment by dhilipkumars** [comment]
@cw-sakamoto Thanks for the contribution.

**Comment by MacTynow** [comment]
@cw-sakamoto I'm curious to know why change the default RBAC value from `false` to `true` when almost all the other charts are the other way around.
There should be consistency between all charts for this kind of things.

**Comment by cw-sakamoto** [comment]
@MacTynow Hi, thanks for your comment.
Did you read [this discussion](https://github.com/kubernetes/charts/pull/2769#pullrequestreview-77692014) ? 

Since [this issue](https://github.com/kubernetes/helm/issues/3098) is agreed between charts-maintainers, Would you tell their your opinion?

>There should be consistency between all charts for this kind of things.

You've got a point.

But as I am in agreement with that issue, and the charts-maintainers also agrees, so I do not think it is not curious that default RBAC value is true.

**Comment by MacTynow** [comment]
Thanks for pointing me to that discussion. I agree that there should be consistency, one way or another doesn't matter for me. I was just surprised because this chart is the first one to be consistent with those new conventions. I will then close my PR.

**Pull request merged**